### PR TITLE
Yrob/browsable api improvements

### PR DESCRIPTION
**Pull request by Yrob**
Update aan browsable api.
* Browsable api ondersteund authoriseren met keycloak
* headers en query parameters kunnen worden gezet met een formuliertje.
* Opties voor headers en query parameters worden gesuggereerd obv de openapi spec.

Parameters:
image

Headers:
image

### CODE REVIEW

File: src/dso_api/settings.py [hunk 1]
```diff
@@ -105,6 +105,7 @@
 
 ROOT_URLCONF = "dso_api.urls"
 
+STATICFILES_DIRS = [str(BASE_DIR / "dso_api/static")]
```
Comment (lbam) [inline src/dso_api/settings.py:108]:
```suggestion
STATICFILES_DIRS = [str(BASE_DIR / "dso_api" / "static")]
```

But shouldn't we call this `MEDIA_ROOT` or `STATIC_ROOT` to conform to [Django conventions](https://docs.djangoproject.com/en/3.2/howto/static-files/)?

Comment (lbam) [reply src/dso_api/settings.py:108]: Oh, we already have `STATIC_ROOT` and we also have `django.contrib.staticfiles` in the `INSTALLED_APPS`. Any reason not to use that for these new files?

**Review by tlrh314 — APPROVED** [review]

File: src/dso_api/static/dso_api/dynamic_api/js/browsable_api.js [hunk 2]
```diff
@@ -0,0 +1,828 @@
...
+  // Get the headers that should be used with the request
+  headers = {};
+  headerSettings.forEach(h => {
+    if (h.active) {
+      headers[h.key] = h.value;
+    }
+  })
+
+  let reqInfoEl = document.getElementById("request-info")
+  reqInfoEl.innerHTML = `<pre ><b>${method}</b> ${PAGEURL}</pre>`
```
Comment (vdboor) [inline src/dso_api/static/dso_api/dynamic_api/js/browsable_api.js:121]:
```suggestion
  reqInfoEl.innerHTML = `<pre><b>${method}</b> ${PAGEURL}</pre>`
```

File: src/dso_api/settings.py (hunk 1 above)
Comment (vdboor) [reply src/dso_api/settings.py:108]: @lbam The `STATIC_ROOT` is where all collected files get *written* to. So you can't use that for providing assets!

**Comment by vdboor** [comment]
Really impressive work!

I would add some comments to the JavaScript, stating why certain functions exist/what they do. I see a lot of things doing something with "parameters" and "settings" which is confusing as first time reader.

File: src/dso_api/settings.py (hunk 1 above)
Comment (lbam) [reply src/dso_api/settings.py:108]: Oh, my bad, misread the docs. Never mind!

File: src/dso_api/static/dso_api/dynamic_api/js/browsable_api.js [hunk 3]
```diff
@@ -0,0 +1,828 @@
...
+  setURL(PAGEURL.href);
+  setParams();
+  setHeaders();
+  setPageLinks(page);
+
+  getData(PAGEURL).catch(parseException).then(data => {
+    if (data) {
+      parseData(data);
+    }
+    window.history.replaceState(JSON.parse(JSON.stringify({
```
[the remark this answers is not in the document]
Comment (Yrob) [reply src/dso_api/static/dso_api/dynamic_api/js/browsable_api.js:67]: Omdat het object dat in de history geschoten wordt dom elementen kan bevatten, als je die direct in de history schiet krijg je een "Dom Object could not be cloned" error.

[the remark this answers is not in the document]
Comment (vdboor) [reply src/dso_api/static/dso_api/dynamic_api/js/browsable_api.js:67]: ok!

**Review by vdboor — APPROVED** [review]

**Pull request merged**